### ISSUE TITLE
Bug in message serialization when using `Node::request`

### ISSUE THREAD

**Issue by ViliamVadocz** [eduidl/gz-rs#16]
I think I discovered a bug in the serialization. This is how I make a request:

```rs
let (res, ok) = node.request::<_, Boolean>(
    "/world/test_world/set_pose",
    Pose {
        name: "my_model".to_string(),
        position: MessageField::some(Vector3d {
            x,
            y,
            z,
            ..Default::default()
        }),
        ..Default::default()
    },
    Duration::from_millis(10),
).unwrap();
```

`x`, `y`, and `z` are `f64` that I can change. Depending on which coordinates I pick, the request either works (sets the correct position) or doesn't (moves the model to the origin instead).

I believe that the request doesn't work if the serialized message (specifically the `Pose`) contains any zero byte.

For example for `x = 0, y = 1.5, z = 0` I get this serialization for `Pose`: `[18, 5, 115, 99, 97, 108, 101, 34, 9, 25, 0, 0, 0, 0, 0, 0, 248, 63]`. This request does not work.

On the other hand, for `x = -0.05, y = 0, z = 1.102` the serialization is `[18, 5, 115, 99, 97, 108, 101, 34, 18, 17, 154, 153, 153, 153, 153, 153, 169, 191, 33, 111, 18, 131, 192, 202, 161, 241, 63]` which does work.

---

I think you assumed that the serialization cannot contain zero bytes when you wrote `Node::request`, seeing as you push a zero onto the end. This assumption is wrong. Maybe you need to sanitize the serialization.

**Comment by eduidl** [comment]
Thank you for your detailed reporting.
I'm fixing it in https://github.com/eduidl/gz-rs/pull/17

**Comment by eduidl** [comment]
Fixed from [v0.7.3](https://crates.io/crates/gz/0.7.3)
